app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -75,29 +84,30 @@ class HomeFragment : Fragment() {
     }
 }
 
-private class HomeTileAdapter(val onUrlEnteredListener: OnUrlEnteredListener) :
+private class HomeTileAdapter(val onUrlEnteredListener: OnUrlEnteredListener, homeTiles: MutableList<JSONObject>, context: Context) :
         RecyclerView.Adapter<TileViewHolder>() {
-
+    val context = context
     val tiles = listOf(
```

Comment:
You're processing each tile again, after pulling it from disk - I think you should just do everything at once, and pass in the list, rather than forming it again.

This code was originally here because we were hard-coding the list here, but we can move this to the init.

Reply:
Optional: also think a little about how you would incorporate #443 - where would you want to process that? When would you need to handle the blacklist?